src/irr/video/COpenGLPipelineCache.cpp

```diff
@@ -3,6 +3,76 @@
 using namespace irr;
 using namespace video;
 
+static int compare_desc_layouts(const IGPUDescriptorSetLayout* A, const IGPUDescriptorSetLayout* B)
+{
+	// a null descriptor set layout is the smallest
+	if (!A)
+		return B ? -1 : 0;
+	if (!B)
+		return A ? 1 : 0;
+
+	// non null descriptor set layouts, we can compare more
+	auto count = A->getBindings().length();
+	auto lendiff = int64_t(count) - int64_t(B->getBindings().length());
+	if (lendiff != 0)
+		return lendiff;
+
+	const auto* lhs = A->getBindings().begin();
+	const auto* rhs = B->getBindings().begin();
```

Comment:
I'd compare the OpenGL multibind arguments here instead, cause after all you only care if the bindings got remapped the same to OpenGL slots (which means that as you iterate through the `(set,binding)` tuples of both layouts, do all 4 running total counts of UBO, SSBO, Texture and Image stay the same?)

But I guess that can be left as a TODO in the comment, since this seems to be "good enough"